### PR TITLE
fix error analysis cohort filter for string label values

### PR DESCRIPTION
**Pull request by imatiach-msft**
## Description

Fix the cohort filter to support string label values.  Otherwise, when creating a filter based on "True Y" or "Predicted Y", and changing the base cohort, error analysis can fail to filter the data correctly because the filter from UI is in terms of indexes while the data itself has string values.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: erroranalysis/tests/test_cohort_filter.py [hunk 1]
```diff
@@ -355,6 +376,19 @@ def create_iris_pandas():
     return X_train, X_test, y_train, y_test, feature_names
 
 
+def create_iris_str_y():
```
Comment (gaugup) [inline erroranalysis/tests/test_cohort_filter.py:379]: could you reuse the create_iris_pandas? Seems like duplicate code here. Maybe pass in a boolean flag in that function to create string labels.

File: erroranalysis/tests/test_cohort_filter.py [hunk 2]
```diff
@@ -83,6 +83,27 @@ def test_cohort_filter_true_y(self):
                            model_task,
                            filters=filters)
 
+    def test_cohort_filter_string_encoded_true_y(self):
```
Comment (gaugup) [inline erroranalysis/tests/test_cohort_filter.py:86]: This test looks very identical to `test_cohort_filter_true_y`. Could you reuse the test `test_cohort_filter_true_y` and parametrize it?

File: erroranalysis/tests/test_cohort_filter.py (hunk 1 above)
Comment (imatiach-msft) [reply erroranalysis/tests/test_cohort_filter.py:379]: done!

File: erroranalysis/tests/test_cohort_filter.py (hunk 2 above)
Comment (imatiach-msft) [reply erroranalysis/tests/test_cohort_filter.py:86]: done, parameterized it

**Review by gaugup — APPROVED** [review]

**Pull request merged**